src/WebJobs.Script/Settings/ScriptSettingsManager.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Microsoft.Azure.WebJobs.Script.Settings
+{
+    public sealed class ScriptSettingsManager : ISettingsManager
```

Comment:
I think it would be helpful to have a second `ISettingsManager` implementation (or derived class if we remove the interface as Mathew mentions). Specifically, I've wanted to be able to have a `devsettings.json` file on disk somewhere that the tests would read from rather than using environment variables. If that file wasn't there, it could fallback to default behavior. It would make new machine setup a lot faster (just copy the file) and allow you to change values in between test runs without needing to restart VS to pick up the new env variables. It would also help validate the patterns.